services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/MultipartManagerV2Impl.java

```diff
@@ -115,16 +116,13 @@ <T extends MultipartRequest> MultipartUploadStatus startOrResumeMultipartRequest
 
 		// The MD5 is used to identify if this upload request already exists for this user.
 		String requestMD5Hex = MultipartRequestUtils.calculateMD5AsHex(request);
-		

```

Comment:
if(!forceRestart && filleHandleExists, but storage object does not exist){
throw new IllegalArgumentException("You are attempting to re-upload a file that was already uploaded as FileHandle.id = 123.  However the file: <bucket>, <key> no longer exits in the cloud storage provider.  Please delete the Original FileHandle and try again.  Or attempt to re-upload again with forceRestart=true.")

We want them to fix the problem, not just silently sweep it under the rug.
}
An alternate solution would be to "fix" the broken file handle (not create a new one).  This would involve letting them re-upload the file, then change the broken file handle to point to the newly uploaded object.  This might be the best option because it allows them to truly fix what they broke.